### PR TITLE
[SCSB-145] require Python 3.10

### PR DESCRIPTION
**Pull request by zacharyburnett**
resolves [SCSB-145](https://jira.stsci.edu/browse/SCSB-145)

propagate Astropy's deprecation of Python 3.9 to downstream packages

> [!NOTE]
> this PR was generated automatically by ``batchpr`` :robot:

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,123 +1,129 @@
...
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.9",
 ]
 dependencies = [
-    'asdf >=2.15.1',
-    'astropy >=5.3',
-    'crds >=10.3.1',
-    'defusedxml >=0.5.0',
-    'galsim >=2.5.1',
-    # 'rad >=0.18.0',
```
Comment (PaulHuwe) [inline pyproject.toml:24]: Can you save the three commented lines here please?

**Review by PaulHuwe — CHANGES_REQUESTED** [review]

**Review by PaulHuwe — APPROVED** [review]
LGTM

**Comment by schlafly** [comment]
Sorry, Zach, we didn't merge this back when you submitted it and now it's decayed.  I tried to merge in updates and resolve the conflicts but it looks like I failed.  I think the only real thing that isn't formatting that it's doing now is adding some py311 runners, but maybe I'm missing something.  Can you figure out what you want to do here?

**Comment by zacharyburnett** [comment]
> Sorry, Zach, we didn't merge this back when you submitted it and now it's decayed. I tried to merge in updates and resolve the conflicts but it looks like I failed. I think the only real thing that isn't formatting that it's doing now is adding some py311 runners, but maybe I'm missing something. Can you figure out what you want to do here?

sure, rebased now onto `main`

**Comment by schlafly** [comment]
Thanks, merging!

**Pull request merged**